src/components/App.jsx

```diff
@@ -0,0 +1,85 @@
+import React, { Component } from 'react';
+import { connect, Provider } from 'react-redux';
+import PropTypes from 'prop-types';
+import { IntlProvider } from 'react-intl';
+import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
+import SiteFooter from '@edx/frontend-component-footer';
+import { fetchUserAccount, UserAccountApiService } from '@edx/frontend-auth';
+
+import apiClient from '../config/apiClient';
+import { handleTrackEvents } from '../analytics';
+import { getLocale, getMessages } from '../i18n/i18n-loader';
+import SiteHeader from './/SiteHeader';
+import ConnectedProfilePage from './ProfilePage';
+
+import HeaderLogo from '../../assets/edx-sm.png';
+import FooterLogo from '../../assets/edx-footer.png';
+import NotFoundPage from './NotFoundPage';
+
+class App extends Component {
+  componentDidMount() {
+    const { username } = this.props;
+    const userAccountApiService = new UserAccountApiService(apiClient, process.env.LMS_BASE_URL);
+    this.props.fetchUserAccount(userAccountApiService, username);
```

Comment:
This has been refactored to just be connected to redux, that way it can get `authentication` state and its `fetchUserAccount` action the "right" way.